### DISCUSSION TITLE
jQuery $ Conflict

### DISCUSSION THREAD

**Discussion by matthewhartman** [parcel-bundler/parcel#7779] [Q&A] [answered]
Hi There,

I have a React / Typescript app (with no jQuery) and I am using Parcel v2. 

I'm currently experiencing a problem where my output bundle from parcel build is changing a function name to `$`.

When I include my SPA on a parent website that has jQuery, the jQuery definition of $ is overridden from the function coming from the parcel built file.

I did some digging through the docs and managed to find a flag of `--global` used in parcel v1
https://parceljs.org/getting-started/migration/#--global

This allows you to name space the entire bundle, but this option got removed? (not sure why).

Just wondering if there is some other option or way of name spacing my bundle to avoid clashes?

Looking forward to hearing back from you soon.

Cheers,

Matt

**Comment by mischnic** [comment]
Can you provide a code sample that declared a global variable named `$`? That shouldn't be happening in the first place.

The `--global` flag isn't/wasn't related to this and wouldn't solve this problem.

**Reply by matthewhartman** [reply]
Yep. I have uploaded a demo repo here.
https://github.com/matthewhartman/parcel-sandbox

**Reply by matthewhartman** [reply]
My understanding is `--global` will wrap the IIFE into a local `var` so functions within the bundle won't leak globally?

**Reply by matthewhartman** [reply]
Dang it - just realised `$` is an internal function available within Chrome and Firefox. I need to try and replicate this sandbox with what's happening in the integrated parent website. I need more time to figure this out and I will come back with a code example.

**Reply by matthewhartman** [reply]
Okay, I managed to replicate it.

During the minification process, I can see that `$` gets defined within the bundle.

```
$="function"==typeof Symbol&&Symbol.iterator;
```

I can't upload this particular app as it's for commercial / work purpose. But i'm going to try and keep stabbing with the sandbox app to try and replicate it.

**Comment by matthewhartman** [comment, marked as answer]
Managed to fix this issue by tacking on `--no-scope-hoist` to the build script.
